src/main/java/me/desht/pneumaticcraft/common/block/SolarCompressorBlock.java

```diff
@@ -0,0 +1,303 @@
+
+    private static final VoxelShape SHAPE = Stream.of(
+            Block.box(0, 1, 4, 16, 16, 12),
+            Block.box(5, 16, 5, 11, 17, 11),
+            Block.box(2, 15, 3, 3, 17, 13),
+            Block.box(13, 15, 3, 14, 17, 13),
+            Block.box(3, 3, 0, 13, 13, 16),
+            Block.box(1, 1, 1, 15, 15, 15),
+            Block.box(0, 0, 0, 16, 1, 16),
+            Block.box(7, 17, 7, 9, 31, 9)
+    ).reduce((v1, v2) -> Shapes.join(v1, v2, BooleanOp.OR)).get();
+
+    @Override
+    public VoxelShape getShape(BlockState state, BlockGetter world, BlockPos pos, CollisionContext context) {
+        if(state.getValue(BOUNDING)) {
+            return Block.box(0,0,0,0,0,0);
+        }
+        else {
+            return SHAPE;
+        }
+    }
+
+    @Override
+    public VoxelShape getCollisionShape(BlockState state, BlockGetter world, BlockPos pos, CollisionContext context) {
+        if(state.getValue(BOUNDING)) {
+            return Block.box(0,0,0,0,0,0);
+        }
+        else {
+            return SHAPE;
+        }
+    }
+
+    @Override
+    protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockState> builder) {
+        super.createBlockStateDefinition(builder);
+        builder.add(BOUNDING);
+    }
+
+    @Nullable
+    @Override
+    public BlockEntity newBlockEntity(BlockPos pPos, BlockState pState) {
+        return new SolarCompressorBlockEntity(pPos, pState);
+    }
+
+    public BlockPos getMainPos(BlockState state, LevelReader world, BlockPos pos) {
+        // Gets position of main from bounding block based on offset
+        if(state.getValue(BOUNDING)) {
+            SolarCompressorBlockEntity bounding = (SolarCompressorBlockEntity) world.getBlockEntity(pos);
+
+            return pos.subtract(bounding.offsetFromMain);
+        }
+
+        // Returns current position as block is main
+        else {
+            return pos;
+        }
+    }
+
+    @Override
+    public boolean canSurvive(BlockState state, LevelReader world, BlockPos pos) {
+        if (state.getValue(BOUNDING)) {
+            return world.getBlockState(getMainPos(state, world, pos)).getBlock() == this;
+        } else {
+            return world.isEmptyBlock(pos.offset(0,1,0)) &&
+                    world.isEmptyBlock(pos.offset(0,1,1)) &&
+                    world.isEmptyBlock(pos.offset(0,1,-1));
+        }
+    }
+
+    @Override
+    public void setPlacedBy(Level world, BlockPos pos, BlockState state, LivingEntity par5EntityLiving, ItemStack par6ItemStack) {
+        super.setPlacedBy(world, pos, state, par5EntityLiving, par6ItemStack);
+
+        // Declares bounding block offsets
+        Vec3i boundingVec1 = new Vec3i(0, 1, 0);
+        Vec3i boundingVec2 = new Vec3i(0, 1, 1);
+        Vec3i boundingVec3 = new Vec3i(0, 1, -1);
+
+        // Creates bounding blocks and sets their offsets
+        world.setBlock(pos.offset(boundingVec1), world.getBlockState(pos).setValue(BOUNDING, true), Block.UPDATE_ALL);
```

Comment:
Uneasy about the unchecked cast here.  I know the block entity *should* always by a solar compressor, but Minecraft has a way of defeating expectations, in my experience...